### PR TITLE
UNI-9224: ProBuilder v5.0 documentation (manual only)

### PR DESCRIPTION
**Pull request by unitALG**
**DO NOT FORGET TO INCLUDE A CHANGELOG ENTRY**

### Purpose of this PR

Here are the updates for the v5.0 documentation. Besides the main features of the Shape tool and the Cut tool, I did the following:

- Added a new page describing the differences between actions and tools.
- "Rebranded" the entire documentation to use 'action' and 'tool' consistently.
- Updated the workflow sections to use the new Shape tool methods (and to a lesser extent, Cut tool).
- Compared and updated the Preferences page to the documentation.
- Aligned the Hotkeys information with what the Shortcuts Manager lists.
- Replaced the docs where it described setting shortcuts on the PB Preferences window with using the Shortcut Manager.
- Updated most of the images for menus and toolbars since they had fallen out of date.
- Documented the Duplicate Faces and Offset Elements actions.
- Removed the Invert Selection documentation.
- Removed the Bezier tool button from the toolbar screenshots and added more warnings about using experimental features.
- Made sure every tool, action, or menu item listed all access points (button, menu item, shortcut key).
- Added a What's New page for v5.0 (this is part of standards compliance) and removed the old document revision table.
- Ran the DocTools in Validation mode and fixed most of the bad links found.

### Links

**Jira:** https://unity3d.atlassian.net/browse/UNI-9224 
This is the main Unity Roadmap task. I added a lot of subtasks to help organize the work.

I uploaded a copy of this version of the package documentation to an internal test server just to give a little context and provide a demonstration of what the final docs will look like. You will need to use a VPN to view them (since the server is internal): http://docs.hq.unity3d.com/wb-team/com.unity.probuilder@5.0/manual/index.html

### Comments to Reviewers

I've finished all of the work except for the API review. 
I haven't touched the **About the ProBuilder Scripting API** page, which provides an overview of the API because I wanted to do the API review first.

Throughout the documentation, I have added some comments, questions, etc. for the reviewers. Please keep an eye out for these messages (marked with the `@DEVNOTE:` prefix, and colored blue). Most of them are just requests for clarification but at least one is concerned with some issues I had with toggling the Experimental Preference.

I can also provide a Google doc if you prefer to leave comments in that way. Please slack me if you want me to provide that.

### CODE REVIEW

File: Documentation~/workflow-create-bezier.md [hunk 1]
```diff
@@ -2,8 +2,18 @@
...
+
+2. Open the ProBuilder window (in Unity's top menu: **Tools** > **ProBuilder window**).
+
+	The [Edit Mode toolbar](edit-mode-toolbar.md) and the [ProBuilder toolbar](toolbar.md) appear.
+
+3. From the ProBuilder toolbar, click ![Bezier Shape icon](images/icons/NewBezierSpline.png) **New Bezier Shape**. 
+
+	> **Tip:** You can also access this tool from the ProBuilder menu (**Tools** > **ProBuilder** > **Editors** > **New Bezier Shape**).
+
+	The initial curve has two control points with tangent handles to control the curve's bend. By default, ProBuilder creates the curve [in editing mode](bezier.md) so you can continue to reshape it.
```
Comment (lopezt-unity) [inline Documentation~/workflow-create-bezier.md:19]: Beziers shapes are referenced in both `Creating meshes` and `Experimental features` sections. Is this desirable ? Or should it be only in one place (I guess experimental one would make more sense)?

File: Documentation~/workflow-edit-tasks.md [hunk 2]
```diff
@@ -28,28 +30,30 @@ This page describes some of the most common ways you can use the ProBuilder tool
...
 
 - **Add a door**: follow the *Make a hole* procedure, but to start, create a [Door shape](Door.md). Fit the hole to match, [merge the two objects](Object_Merge.md), and then [weld the door to the frame](Vert_Weld.md) you created.
 
-- **Make a tunnel with normals on both the inside and outside**: duplicate the tunnel object and scale it slightly so that you [bridge the edges](Edge_Bridge.md). Next, [flip the normals](Object_FlipNormals.md) on the smaller one. 
+- **Make a tunnel with normals on both the inside and outside**: duplicate the tunnel object and scale it slightly so that you [bridge the edges](Edge_Bridge.md). Next, [flip the normals](Object_FlipNormals.md) on the smaller one.
 
 - **Make a building with towers**: start with a [cube](Cube.md) and [inset](Face_Inset.md) on the top face, then [extrude](Face_Extrude.md) upward. Repeat this as many times as you like. You could also [subdivide the top face](Face_Subdivide.md) to create multiple extrusions.
 
+<span style="color:blue">@DEVNOTE: Do you have some good examples for using the Cut tool to make either objects or buildings for this section here? </span>
+
```
Comment (lopezt-unity) [inline Documentation~/workflow-edit-tasks.md:56]: It's a really good tool to make custom cuts and broke flat surfaces in smaller ones. 
But @gabrielw-us will have more examples on this I guess :)

File: Documentation~/Vert_Connect.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+The __Connect Vertices__ action creates a new edge that connects the selected vertices.
 
 ![Insert a new edge between two vertices on a face](images/ConnectVerts_Example.png)
 
 
 
-> **Tip:** You can also use this tool with the **Alt/Opt+E** hotkey.
+> **Tip:** You can also use this action with the **Alt/Opt+E** hotkey, or from the ProBuilder menu (**Tools** > **ProBuilder** > **Selection** > **Smart Connect**).
 
 If you select more than two vertices, ProBuilder creates as many new edges as possible, and adds extra vertices where necessary in order to keep the geometry valid. For example, if you connect three vertices around a quad, ProBuilder creates a new vertex in the middle to support the three new edges.
```
Comment (karl-) [inline Documentation~/Vert_Connect.md:11]: recommend "creates as many new edges as required"

**Review by karl- — APPROVED** [review]

File: Documentation~/whats-new.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+* The new [Shape](shape-tool.md) tool has been completely rebuilt as an [EditorTool](https://docs.unity3d.com/ScriptReference/EditorTools.EditorTool.html). 
+* The [Poly Shape](polyshape.md) tool is now an [EditorTool](https://docs.unity3d.com/ScriptReference/EditorTools.EditorTool.html).
+
+### Fixed
+
+* Fixed Undo completely reverting all actions.
+
+For a full list of changes and updates in this version, see the [ProBuilder package changelog](../changelog/CHANGELOG.html).
+
+<span style="color:blue">@DEVNOTE: We are now supposed to include a **What's new** page for every major release of packages. This is supposed to be a quick overview of the major new and improved features, and any important bug fixes. Is there anything you feel should be highlighted here? </span>
```
[the remark this answers is not in the document]
Comment (karl-) [reply Documentation~/whats-new.md:22]: Good suggestions, Thomas. Regarding Select Path tool, I think this was already published but it's probably worth including regardless.

File: Documentation~/menu.md [hunk 5]
```diff
@@ -1,9 +1,11 @@
 # The ProBuilder menu
 
-Use the ProBuilder menu to access most of the ProBuilder editors and tools, and the special tools that are only available through this menu. For example, you can access tools to repair and debug ProBuilder, or access the experimental [Boolean](boolean.md) tool. 
+Use the ProBuilder menu to access most of the ProBuilder editors, actions, tools, and a few special features that are only available through this menu. For example, you can repair and debug ProBuilder, or access the experimental [Boolean](boolean.md) Editor window. 
 
 ![The ProBuilder menu](images/pb_menu.png)
 
+<span style="color:blue">@DEVNOTE: Is the **Tools > Debug** menu entry permanent? If so, can you give me some context for these? Who are the users? What would you do with them? </span>
```
[the remark this answers is not in the document]
Comment (karl-) [reply Documentation~/menu.md:7]: That menu is not available in published builds, only dev branches.

File: Documentation~/menu-editors.md [hunk 6]
```diff
@@ -1,26 +1,26 @@
 # Editors
 
-This sub-menu provides access to editor and component windows.
+This sub-menu provides access to tools and editor windows.
 
 ![Tools > ProBuilder > Editors menu](images/menu-editors.png)
```
[the remark this answers is not in the document]
Comment (karl-) [reply Documentation~/menu-editors.md:5]: Yes probably, I thought they used the same priority constants... ? Maybe something got crossed to break that.

File: Documentation~/whats-new.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (amylouisee) [reply Documentation~/whats-new.md:22]: I think you have the wrong amylouisee

[the remark this answers is not in the document]
Comment (unitALG) [reply Documentation~/whats-new.md:22]: Hi, unitALG is the amylouise you are looking for :-)
As for SelectPath, I haven't seen that tool. I couldn't find it on the toolbar, menus, or Shortcuts Manager.
Can you give me an overview (maybe offline) please?

File: Documentation~/menu.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (unitALG) [reply Documentation~/menu.md:7]: I pulled from the master branch. Was it removed on the 5.0 release branch? I'll check that out...

File: Documentation~/workflow-create-bezier.md (hunk 1 above)
Comment (unitALG) [reply Documentation~/workflow-create-bezier.md:19]: Good point. I'll make the changes.

File: Documentation~/Cube.md [hunk 7]
```diff
@@ -1,17 +1,8 @@
 # Cube
-A cube is the default shape in ProBuilder. It is a six-sided 3D square. To add a default cube to the Scene, you can use the **Ctrl/Cmd+K** keyboard shortcut.
+A cube is the default shape in ProBuilder. It is a six-sided 3D square.
```
[the remark this answers is not in the document]
Comment (unitALG) [reply Documentation~/Cube.md:2]: It's not in the Shortcuts Manager and it doesn't work when I try it.

File: Documentation~/menu.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (unitALG) [reply Documentation~/menu.md:7]: It's fine in the 5.0.0-pre.12 build published to the Candidates Registry. Thanks!

File: Documentation~/Cube.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (unitALG) [reply Documentation~/Cube.md:2]: Same with v5.0.0-pre.12

File: Documentation~/Object_ProBuilderize.md [hunk 8]
```diff
@@ -1,18 +1,17 @@
...
+The __ProBuilderize__ action converts the selected object(s) into ProBuilder-editable objects.
 
+> **Tip:** You can also launch this action from the ProBuilder menu (**Tools** > **ProBuilder** > **Object** > **ProBuilderize**).
 
 ## ProBuilderize options
 
 ![ProBuilderize options](images/Object_ProBuilderize_props.png)
 
-
+<span style="color:blue">@DEVNOTE: What is the **Preserve Faces** option? It is mentioned in the info box on the options</span>
```
[the remark this answers is not in the document]
Comment (unitALG) [reply Documentation~/Object_ProBuilderize.md:11]: The problem is that I can't figure out where it is. It doesn't seem to be one of the ProBuilderize options nor is it a Preference (not that I can see).

File: Documentation~/whats-new.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (unitALG) [reply Documentation~/whats-new.md]: Added documentation for the Select Path tool in the last changeset, including a mention on the Shortcuts page and the What's New page, and the Action reference page.

**Pull request closed without merging**